crates/fuel-core/src/query/subscriptions.rs

```diff
@@ -60,11 +61,15 @@ where
             match status {
                 TxStatusMessage::Status(status) => {
                     let status = ApiTxStatus::new(transaction_id, status);
-                    Ok(status)
+                    if !allow_preconfirmation && (matches!(status, ApiTxStatus::PreconfirmationFailure(_)) || matches!(status, ApiTxStatus::PreconfirmationSuccess(_))) {
```

Comment:
We could add a method for the `status` like `is_preconfirmation` and call it here, will be much easier to read=)

Reply:
The same we can apply to all places where we use `matches!`=)